compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -289,6 +289,17 @@ impl<'a> FunctionContext<'a> {
         self.builder.insert_binary(positive_predicate, BinaryOp::Add, negative_predicate)
     }
 
+    /// Insert constrains ensuring that the operation does not overflow the bit size of the result
+    ///
+    /// If the result is unsigned, we simply range-checks against bit size
+    ///
+    /// If the result is signed, we just prepare it for check_signed_overflow() by casting it to
+    /// an unsigned value representing the signed integer.
+    /// We need to use a bigger bitsize depending on the operation, in case the operation does overflow,
+    /// Then, we delegate the overflow checks to check_signed_overflow() and cast the result back to its type.
+    /// Note that we do NOT want to check for overflows here, only check_signed_overflow() is allowed to do so
+    /// this is because an overflow might be valid. For instance if 'a' is a signed integer, then 'a - a', as an unsigned result, always
+    /// overflow the bit size but the operation is valid (i.e it is not a signed overflow)
```

Comment:
```suggestion
    /// Note that we do NOT want to check for overflows here, only check_signed_overflow() is allowed to do so.
    /// This is because an overflow might be valid. For instance if 'a' is a signed integer, then 'a - a' as an unsigned result will always
    /// overflow the bit size, however the operation is still valid (i.e it is not a signed overflow)
```

Reply:
Done